x/campaign/types/params_test.go

```diff
@@ -18,23 +18,43 @@ func TestParamsValidateBasic(t *testing.T) {
 		err    error
 	}{
 		{
-			name:   "should allow validate valid params",
-			params: NewParams(DefaultMinTotalSupply, DefaultMaxTotalSupply, DefaultCampaignCreationFee),
+			name: "should allow validate valid params",
+			params: NewParams(
+				DefaultMinTotalSupply,
+				DefaultMaxTotalSupply,
+				DefaultCampaignCreationFee,
+				DefaultMaxMetadataLength,
+			),
 		},
 		{
-			name:   "should prevent validate invalid min total supply",
-			params: NewParams(sdkmath.ZeroInt(), DefaultMaxTotalSupply, DefaultCampaignCreationFee),
-			err:    errors.New("minimum total supply should be greater than one: invalid total supply range"),
+			name: "should prevent validate invalid min total supply",
+			params: NewParams(
+				sdkmath.ZeroInt(),
+				DefaultMaxTotalSupply,
+				DefaultCampaignCreationFee,
+				DefaultMaxMetadataLength,
+			),
+			err: errors.New("minimum total supply should be greater than one: invalid total supply range"),
 		},
 		{
-			name:   "should prevent validate min total supply greater than max",
-			params: NewParams(DefaultMaxTotalSupply, DefaultMinTotalSupply, DefaultCampaignCreationFee),
-			err:    errors.New("maximum total supply should be greater or equal than minimum total supply: invalid total supply range"),
+			name: "should prevent validate min total supply greater than max",
+			params: NewParams(
+				DefaultMaxTotalSupply,
+				DefaultMinTotalSupply,
+				DefaultCampaignCreationFee,
+				DefaultMaxMetadataLength,
+			),
+			err: errors.New("maximum total supply should be greater or equal than minimum total supply: invalid total supply range"),
 		},
 		{
-			name:   "should prevent validate invalid coins for campaign creation fee",
-			params: NewParams(DefaultMinTotalSupply, DefaultMaxTotalSupply, sdk.Coins{sdk.Coin{Denom: "foo", Amount: sdkmath.NewInt(-1)}}),
-			err:    errors.New("coin -1foo amount is not positive"),
+			name: "should prevent validate invalid coins for campaign creation fee",
+			params: NewParams(
+				DefaultMinTotalSupply,
+				DefaultMaxTotalSupply,
+				sdk.Coins{sdk.Coin{Denom: "foo", Amount: sdkmath.NewInt(-1)}},
+				DefaultMaxMetadataLength,
+			),
+			err: errors.New("coin -1foo amount is not positive"),
```

Comment:
We should add a test case in `ValidateBasic` for metadata length too big

Reply:
The `validateMaxMetadataLength` only checks if the type is right, not if the length is too big. Should we define a maximum length for the user can set thought params?

Reply:
Yes right, this is the validity of the new value to be provided that is checked.

What I meant is `ValidateBasic` check `validateMaxMetadataLength`, we should have a test case that verify `ValidateBasic` upon failure on `validateMaxMetadataLength`

Reply:
But if we create a validation regarding the `MaxMetadataLength` should have a max to be set, it makes sense to test the failure cases for `validateMaxMetadataLength`. IMHO don't make sense put the `MaxMetadataLength` as a param if we still have a hardcoded value limiting the maximum

Reply:
Yes, it's true we can't define invalid test case as the type is already cast in the call